pkg/scheduler/factory/plugins.go

```diff
@@ -632,3 +640,19 @@ func ListAlgorithmProviders() string {
 	sort.Strings(availableAlgorithmProviders)
 	return strings.Join(availableAlgorithmProviders, " | ")
 }
+
+// RegisterPredicateToFrameworkConfigProducer registers a framework config producer for a predicate.
+func RegisterPredicateToFrameworkConfigProducer(name string, producer FrameworkConfigProducer) {
+	schedulerFactoryMutex.Lock()
```

Comment:
These 2 functions can be refactored to call a non-exported function that takes the map as argument.

Reply:
Done.